### ISSUE TITLE
After 7 days the app is installed i get the rating popup everytime

### ISSUE THREAD

**Issue by davidBarbieri** [Skyost/rate_my_app#13]
After 7 days the app is installed i get the rating popup everytime i switch page/open the app

This is my init Code 

'''
RateMyApp(
        minDays: 7,
        minLaunches: 10,
        remindDays: 7,
        remindLaunches: 10, 
      ));
'''

This is the way i open the popup:

'''
_rateMyApp.init().then((_) {
      if (_rateMyApp.shouldOpenDialog) {
        _rateMyApp.showStarRateDialog(
          context,
          title: ToTLocalizations.of(context).rateAppTitle,
          message: ToTLocalizations.of(context).rateAppContent,
          ignoreIOS: false,
          starRatingOptions: StarRatingOptions(),
          onRatingChanged: (stars) {
            return [
              RaisedButton(
                padding: EdgeInsets.all(10.0),
...
'''

The expected behavior is that if i click "later" the "baseLaunchDate" should be set to today, and the popup opened again in 7 (remind days) days

I'm running it on IOS 12.4 simulator

The native popup is opened, but as far i can understand there is no management in the plugin for the native form callback up to the dart code and shared preferences.

**Comment by davidBarbieri** [comment]
Thanks for the answer. Something is not clear to me tho.

My use case is:

IOs: native dialog
Android: startRatingDialog


I could use the Platform object, and in case of android do a call to you starRatingDialog plugin and that is great.
But then how could i handle the native rating dialog in ios, using the same configuration (minDays, minLaunces ecc...)? Would be great if your plugin interfaced the callbacks of that popup and integrate it into the same logic (maybe it does but i'm missing it).


Thanks!

David

**Comment by Skyost** [comment]
`showStarRateDialog` can't integrate the same logic because it's not the plugin who are creating user buttons. It's you (based on the rating). But what you can do in `showStarRateDialog` is to return those buttons (in `onRatingChanged` of course) :

```dart
[
  FlatButton(
    child: Text('RATE'),
    onPressed: () {
      _rateMyApp.doNotOpenAgain = true;
      _rateMyApp.save().then((v) {
        Navigator.pop(context);
        // Call _rateMyApp.launchStore(); or whatever you want.
      });
    },
  ),
  FlatButton(
    child: Text('LATER'),
    onPressed: () {
      _rateMyApp.baseLaunchDate = _rateMyApp.baseLaunchDate.add(Duration(
        days: _rateMyApp.remindDays,
      ));
      _rateMyApp.launches -= _rateMyApp.remindLaunches;
      _rateMyApp.save().then((v) => Navigator.pop(context));
    },
  ),
  FlatButton(
    child: Text('NO'),
    onPressed: () {
      _rateMyApp.doNotOpenAgain = true;
      _rateMyApp.save().then((v) => Navigator.pop(context));
    },
  ),
]
```

**Comment by davidBarbieri** [comment]
Yeah the part managed by flutter and dart is clear, but i wanted to keep the native dialog for IOs (as far as i know there is no native for android) and be able to managed it's result thought a method channel future, containing the result of the native popup (true for rate, and false of "later" or click outside), Don't even know if that's possible, but it would create a seamless integration covering all cases with a single implementation.

This is the part of code that opens the native popup, but there is no await for the result of the call, maybe it's possible/easy to add it?

```
if (!ignoreIOS && Platform.isIOS && await _CHANNEL.invokeMethod('canRequestReview')) {
      return _CHANNEL.invokeMethod('requestReview');
    }
```

This is the oBjc code

```
if ([call.method isEqualToString:@"requestReview"]) {
    [SKStoreReviewController requestReview];
    result(nil);
  }
'''

**Comment by davidBarbieri** [comment]
Nevermind, i just saw athat SKStoreReviewController.requestReview() return null, so there is no way to intercept it :/

**Comment by Skyost** [comment]
> Yeah the part managed by flutter and dart is clear, but i wanted to keep the native dialog for IOs (as far as i know there is no native for android) and be able to managed it's result thought a method channel future, containing the result of the native popup (true for rate, and false of "later" or click outside), Don't even know if that's possible, but it would create a seamless integration covering all cases with a single implementation.

Ow okay ! But as you said above, it's not possible because of how Apple designed their dialog.

**Comment by davidBarbieri** [comment]
I'll probably go for full plugin management then !

Thanks for your help!!
